File: Python/tree_mem.py
ONE_BILLION = 1000000000
NUM_TRIALS = 50
cumulative = 0

class Tree:
    def __init__(self, nodeNum):
        self._root = Node(None)
        node_queue = []
        node_queue.append(self._root)
        for i in range(nodeNum):
            current_node = node_queue.pop(0)
            current_node._left_child = Node(current_node)
            current_node._right_child = Node(current_node)
            node_queue.append(current_node._left_child)
            node_queue.append(current_node._right_child)


class Node:
    def __init__(self, p):
        self._left_child = None
        self._right_child = None
        self._parent = p
        self._data = "a"
    


tree = Tree(100000)
print("done")
input()

File: C++/tree_mem.cpp
#include <chrono>
#include <iostream>
#include <unistd.h>
#include <queue>

using namespace std;

class Node {
public:
    Node(Node* p) {
        parent = p;
        this->data = "a";
        this->leftChild = nullptr;
        this->rightChild = nullptr;
    }

    Node* parent;
    Node* leftChild;
    Node* rightChild;
    string data;

    void DFS() {
        data += 1;
        if (leftChild != nullptr) { leftChild->DFS(); }
        if (rightChild != nullptr) {rightChild->DFS(); }
    }
};

class Tree {
public:
    Tree(int nodeNum) {
        rootNode = new Node(nullptr);
        queue<Node*> nodeQueue;
        nodeQueue.push(rootNode);
        for (int i = 0; i < nodeNum; i++) {
            Node* currentNode = nodeQueue.front();
            nodeQueue.pop();
            currentNode->leftChild = new Node(currentNode);
            currentNode->rightChild = new Node(currentNode);
            nodeQueue.push(currentNode->leftChild);
            nodeQueue.push(currentNode->rightChild);

        }
    }

    void DFS() {
        rootNode->DFS();
    }
private:
    Node* rootNode;
};


int main() {



    const int NUM_TRAILS = 50;
    const int ONE_BILLION = 1000000000;
    const int ONE_MILLION = 1000000;
    
    Tree* tree = new Tree(100000);
    cout << "done" << endl;
    string temp;
    cin >> temp;
}

